packages/amplify-e2e-core/src/init/amplifyPush.ts

```diff
@@ -43,12 +45,18 @@ export function amplifyPush(cwd: string, testingWithLatestCodebase = false): Pro
         }
       });
     // Test amplify push
-    spawn(getCLIPath(testingWithLatestCodebase), ['push'], { cwd, stripColors: true, noOutputTimeout: pushTimeoutMS })
+    const pushCommands = spawn(getCLIPath(testingWithLatestCodebase), ['push'], { cwd, stripColors: true, noOutputTimeout: pushTimeoutMS })
       .wait('Are you sure you want to continue?')
-      .sendConfirmYes()
-      .wait('Do you want to generate code for your newly created GraphQL API')
-      .sendConfirmNo()
-      .wait(/.*/)
+      .sendConfirmYes();
+
+    if (!settings?.skipCodegen) {
+      pushCommands.wait('Do you want to generate code for your newly created GraphQL API')
```

Comment:
There is an existing method already for this called `amplifyPushWithoutCodegen` that's used in other tests.

Reply:
Not sure why we have a separate method for these minor differences. Still I feel like this is a good addition as we can eventually get rid of the `amplifyPushWithoutCodegen` with this change.

Reply:
Can you just add a task for this to migrate the usage of `amplifyPushWithoutCodegen`? Don't really want two methods that do the same thing.